ALMA-128: Add holdings display note for CAAS

ztucker4 opened the pull request:
Implements [ALMA-128](https://jira.library.ucla.edu/browse/ALMA-128)

Adds two new functions to the Alma API client to support holdings records GET and PUT, and uses these in a script to add an 852 $z note on CAAS holdings.

The Analytics Report data is limited to the relevant library's holdings (confirmed via another report to never contain 852 $z), so the script simply inserts this subfield on all records. Tested in Sandbox on one holdings record, not yet tested with report integration in Production.

akohler reviewed (changes requested):
@ztucker4 In general I think this is OK, but a few things should change so we can test this completely.
* The Analytics report doesn't have a column for the bib (MMS) id.  
* There are several dozen CAAS holdings records which *do* have an existing 852 $z.  Sandbox example: [holdings 22526317580006533](https://ucla-psb.primo.exlibrisgroup.com/permalink/01UCS_LAL/17p22dp/alma9964268043606533)
  * Adding another 852 $z with the note is fine, but should be tested
* The test data should include a non-existent or invalid bib to confirm the check and error count code works.

Thanks --Andy

ztucker4 commented:
Hi @akohler, I've fixed the issues you mentioned. The report now correctly contains a MMS ID column, I've added an invalid MMS and Holding Id to the sandbox test data (and fixed the logic so that the error checking actually works), and the script now inserts the new 852 $z in the first position. I've asked Stacy to confirm that adding the note in the first position is the desired behavior, so I may need to change that based on her response. Other than that, I believe this is now ready to go.

akohler commented:
@ztucker4 After thinking more about this, I decided it was important to add the new $z in the "right" place: either at the end of the 852 (if no $z already), or before the first existing $z (for the best Primo display).  Putting a $z first in the 852 is not "wrong", but it's not standard.  `pymarc` makes me want to not make the effort... but my sense of MARC morality won out.

I've pushed a commit which adds `get_subfield_position()` along with a new sample record which doesn't already have an 852 $z. (I edited the two sandbox records you've been using, to remove the $z which got added.)

I also added a little bit of debugging code via `pprint` and some reformatting of the XML Alma spits out.  And I commented out the line which does the actual update, to make it easier to "test".

Please take a look, pull this down and try this out; feel free to tweak as needed.  If this approach is OK with you, push final changes to clean up and I'll merge.

Thanks --Andy
